File: src/main/java/com/hf/hfw/api/v1/ReportService.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.hf.hfw.api.v1;

import com.hf.hfw.reports.ReportData;
import com.hf.homefinanceshared.Account;
import java.util.List;
import javax.ws.rs.Consumes;
import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.PathParam;
import javax.ws.rs.Produces;

/**
 *
 * @author pldorrell
 */
public interface ReportService {
    @GET
    @Produces({"application/json"})
    @Consumes({"application/json"})
    @Path("/{accountId}/{reportType}")    
    public ReportData getReport(@PathParam("accountId") String accountId,@PathParam("reportType") String reportType);

    @GET
    @Produces({"application/json"})
    @Consumes({"application/json"})
    @Path("/{accountId}/{reportType}/{reportPeriod}")    
    public ReportData getReport(@PathParam("accountId") String accountId,@PathParam("reportType") String reportType,@PathParam("reportPeriod") String reportPeriod);

    @GET
    @Produces({"application/json"})
    @Consumes({"application/json"})
    @Path("/{accountId}/{reportType}/{startDate}/{endDate}")    
    public ReportData getReport(@PathParam("accountId") String accountId,@PathParam("reportType") String reportType,@PathParam("startDate") String startDate,@PathParam("endDate") String endDate);

}

File: src/main/webapp/script/app/services/reportservice.js
angular.module('ReportServiceModule',[]);

angular.module('ReportServiceModule').service("ReportService", function ($http) {
    //var users = ["Peter", "Daniel", "Nina"]
    var ReportService = {};

    ReportService.getReport = function (reportType) {
        return $http({
            method: 'GET',
            url: '/services/api/v1/reports/' + reportType
        });
    };
    ReportService.getReportForPeriodForAccount = function (accountId, reportType, reportPeriod) {
        return $http({
            method: 'GET',
            url: '/services/api/v1/reports/' + accountId + '/' + reportType + "/" + reportPeriod
        });
    };

    ReportService.getReportForCustomPeriod = function (reportType, startDate, endDate) {
        return $http({
            method: 'GET',
            url: '/services/api/v1/reports/' + reportType + "/" + startDate + "/" + endDate
        });
    };


    return ReportService;
});
